### PR TITLE
Target 2022.1

### PR DESCRIPTION
**Pull request by maartenba**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/coreTools/FunctionsCoreToolsManager.kt [hunk 1]
```diff
@@ -115,16 +115,24 @@ object FunctionsCoreToolsManager {
...
-                            // Ideally match requested size, if not, fall back to non-matching.
-                            // This is the case in the feed for e.g. v2, where Windows has only full size
-                            if (it.size.equals(azureCoreToolsFeedReleaseFilter.size, ignoreCase = true)) 0 else 1
-                        } ?: continue
+                        .sortedWith(compareBy<ReleaseCoreTool> {
+                            // Then match architecture. Items higher in the list have precedence.
+                            // Use 9999 when no match is found.
+                            azureCoreToolsFeedReleaseFilter.architectures.indexOfFirst { architecture ->
+                                it.architecture.equals(architecture, ignoreCase = true)
+                            }.let { rank -> if (rank >= 0) rank else 9999 }
```
Comment (sdubov) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/coreTools/FunctionsCoreToolsManager.kt:127]: Here we have items without a rank moved to the top, if I understand it correctly. Is it what we expect?

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/coreTools/FunctionsCoreToolsManager.kt [hunk 2]
```diff
@@ -115,16 +115,24 @@ object FunctionsCoreToolsManager {
...
+                        }.thenBy {
+                            // Then match size. Items higher in the list have precedence.
+                            // Use 9999 when no match is found.
+                            azureCoreToolsFeedReleaseFilter.sizes.indexOfFirst { size ->
+                                it.size.equals(size, ignoreCase = true)
+                            }.let { rank -> if (rank >= 0) rank else 9999 }
```
Comment (sdubov) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/coreTools/FunctionsCoreToolsManager.kt:133]: Same is here.

File: PluginsAndFeatures/azure-toolkit-for-intellij/gradle.properties [hunk 3]
```diff
@@ -5,7 +5,7 @@ org.gradle.jvmargs='-Duser.language=en'
 sources=false
 
 intellij_version=IC-221.5080-EAP-CANDIDATE-SNAPSHOT
-rider_version=RD-2022.1-EAP9-SNAPSHOT
+rider_version=RD-2022.1-EAP10-SNAPSHOT
```
Comment (sdubov) [inline PluginsAndFeatures/azure-toolkit-for-intellij/gradle.properties:8]: I guess we would need to bump the version to the release SDK when it is published.

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/coreTools/FunctionsCoreToolsManager.kt (hunk 1 above)
Comment (maartenba) [reply PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/coreTools/FunctionsCoreToolsManager.kt:127]: This is indeed expected.

Comment (maartenba) [reply PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/coreTools/FunctionsCoreToolsManager.kt:127]: If the requirement is:
* Windows - x64 - minified -> rank 0
* Windows - x64 - full -> rank 1

Then:
* Windows - x86 - full -> rank 9999

Using `sortedBy()`, that gives rank 0 highest priority (and 9999 lowest)

**Review by sdubov — APPROVED** [review]

**Pull request merged**